### PR TITLE
Sprint 43.5

### PR DESCRIPTION
**Pull request by ericsorenson**
Notable changes:
* Terraform has been upgraded to 0.12
* Event codes have been refactored, requiring a reset of the cases in DynamoDB and ElasticSearch
* The bundle build step requirement needs to be removed

### CODE REVIEW

**Comment by kfoley-18F** [comment]
We use this list when performing a code review to ensure that all tasks have been completed.

- [x] begin by deleting any existing copy of the Git repository, running `docker system prune` to remove any cached artifacts from any prior code review
- [x] clone a fresh copy of the Git repository (`git clone https://github.com/ustaxcourt/ef-cms.git`), fetch the pull request and name it for the sprint (e.g., `git fetch origin pull/204/head:sprint-43`), and then switch to that branch (e.g. `git checkout sprint-43`)
- [x] review the pull request itself, to get oriented
	- [x] read the description of the pull request, which should summarize the changes made
	- [x] read through every task on the Scrum board that's encompassed by this pull request
	- [x] read the description of the commits that comprise the pull request
- [x] stand up the site locally, with `./docker-run.sh`
	- [x] test all functionality in all major browsers, emphasizing the functionality that this pull request addresses
		- [x] for internal Court functionality, perform the most thorough testing in Chrome, though also test in Edge and Firefox
		- [x] for public-facing functionality, test in browsers consistent with [public browser use data](https://analytics.usa.gov/), including Internet Explorer 11
		- [x] test in Mobile Safari and Mobile Chrome, with the caveat that not all internal Court functionality will be necessary on these platforms
	- [x] use an automated audit tool for code quality and practices (recommended: [Chrome DevTools](https://developers.google.com/web/tools/chrome-devtools/), aka [Lighthouse](https://chrome.google.com/webstore/detail/lighthouse/blipmdconlkpinefehnmjammfjpmpbjk))
		- [x] look at efficiency of page loads, asset sizes, HTTP connection management, etc.
	- [x] review for accessibility
		- [x] use an automated audit tool, such as Chrome Audit or aXe
		- [x] navigate site only with the keyboard
		- [x] use VoiceOver or Narrator to navigate the site with audio only, with the display turned off
		- [x] manually test anything that pa11y cannot test automatically (e.g., contrast of text over images)
- [x] [review static code analysis results](https://sonarcloud.io/organizations/flexion-github/projects) in the vendor’s SonarCloud environment
- [x] examine OWASP ZAP output in `docs/`, to ensure that any errors are known to be false positives or have been previously declared to be acceptable
- [x] skim all new code, in the context of existing code, [looking for problems](#what-we-look-for) (knowing that the vast majority of new code will be covered by tests)
- [x] review all tests
	- [x] look at code coverage of tests in [the vendor's SonarCloud environment](https://sonarcloud.io/organizations/flexion-github/projects)
	- [x] methodically review all new tests for correctness, quality of naming
- [x] determine what code isn’t tested, review that rigorously
- [x] review documentation to ensure that it matches changes
- [x] provide comments on the pull request on GitHub, as necessary
	- [x] for comments that are specific to a particular line of code, comment on those specific lines
	- [x] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments
- [x] for each feature-level bug (i.e., it’s working as designed, but designed wrong), open a new issue and put it in the backlog

**Comment by kfoley-18F** [comment]
I was testing on my local machine Creating a new case as a petitioner, using Safari. Ran into an "unexpected error"
Screen Shot 2020-07-10 at 5 32 27 PM

Stack dump and additional info is in slack: https://ustaxcourt.slack.com/archives/CD58Y6799/p1594427623159200

File: shared/src/business/entities/cases/Case.test.js [hunk 1]
```diff
@@ -48,11 +49,108 @@ describe('Case entity', () => {
...
+    });
+  });
+
+  describe('conditionally sets userId on entity', () => {
+    it('sets userId to current user if current user matches rawCase', () => {
+      const myCase = new Case(
+        { ...MOCK_CASE, userId: applicationContext.getCurrentUser().userId },
+        { applicationContext },
+      );
+      expect(myCase.userId).toEqual(applicationContext.getCurrentUser().userId);
```
Comment (julialeague) [inline shared/src/business/entities/cases/Case.test.js:141]: @ericsorenson should `myCase.toRawObject.userId` be the result being checked in this and the test below?

File: shared/src/business/entities/cases/Case.test.js [hunk 2]
```diff
@@ -124,6 +220,16 @@ describe('Case entity', () => {
       expect(Object.keys(myCase)).not.toContain('associatedJudge');
     });
 
+    it('does not create a secondary contact when one is not needed by the party type', () => {
```
Comment (julialeague) [inline shared/src/business/entities/cases/Case.test.js]: @ericsorenson I'm not clear on why this is located in a describe block for "filtered" - is that intentional?

File: shared/src/business/useCases/docketEntry/fileCourtIssuedDocketEntryInteractor.js [hunk 3]
```diff
@@ -158,20 +167,34 @@ exports.fileCourtIssuedDocketEntryInteractor = async ({
...
-      applicationContext,
-      workItem: workItem.validate().toRawObject(),
-    }),
     applicationContext.getPersistenceGateway().updateCase({
       applicationContext,
       caseToUpdate: caseEntity.validate().toRawObject(),
     }),
   ];
 
+  if (isUnservable) {
```
Comment (julialeague) [inline shared/src/business/useCases/docketEntry/fileCourtIssuedDocketEntryInteractor.js:175]: @ericsorenson Should we add a new test for this? I didn't see one added to the test file.

File: shared/src/business/utilities/htmlGenerator/index.pug [hunk 4]
```diff
@@ -13,23 +13,4 @@ html(lang="en")
...
-        .case-information
-          #caption #{content.caseCaptionWithPostfix}
-          #docket-number Docket Number #{content.docketNumberWithSuffix}
-          .clear
-
-        if options.h3
-          h3= options.h3
-
-        | !{content.main}
+      | !{content}
```
Comment (julialeague) [inline shared/src/business/utilities/htmlGenerator/index.pug:16]: @ericsorenson new line? 😸

File: shared/src/business/utilities/pdfGenerator/documentTemplates/NoticeOfTrialIssued.test.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    judge: 'Judge Dredd',
+    postalCode: '80008',
+    startDate: '02/02/2020',
+    startTime: '9:00 AM',
+    state: 'ZZ',
+  };
+
+  it('renders a document header with trial information', () => {
+    const wrapper = mount(
+      <NoticeOfTrialIssued
```
Comment (julialeague) [inline shared/src/business/utilities/pdfGenerator/documentTemplates/NoticeOfTrialIssued.test.js:21]: @ericsorenson This is reused in most of the tests here - could it be pulled up into the describe block?

File: web-api/verify-private-elasticsearch.sh [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+ENV=$1
+
+endpoint=$(aws es describe-elasticsearch-domain --domain "efcms-search-$ENV" --region us-east-1 | jq -r ".DomainStatus.Endpoint" )
+
+response=$(curl -I "https://$endpoint" | head -n 1 | cut -d$' ' -f2)
+
+if [[ "$response" != "403" ]]; then
+  echo "ERROR: expected the elasticsearch endpoint of $endpoint to be private"
+  exit 1
+fi
```
Comment (julialeague) [inline web-api/verify-private-elasticsearch.sh]: @ericsorenson new line? 😸

File: web-api/verify-private-s3-buckets.sh [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  blockPublicPolicy=$(echo "${blocks}" | jq -r ".PublicAccessBlockConfiguration.BlockPublicPolicy")
+  restrictPublicBuckets=$(echo "${blocks}" | jq -r ".PublicAccessBlockConfiguration.RestrictPublicBuckets")
+
+  [ "${blockPublicAcls}" != "true" ] && echo "ERROR: the bucket of $bucket is not fully private: BlockPublicAcls must be set to true" && exit 1
+  [ "${ignorePublicAcls}" != "true" ] && echo "ERROR: the bucket of $bucket is not fully private: IgnorePublicAcls must be set to true" && exit 1
+  [ "${blockPublicPolicy}" != "true" ] && echo "ERROR: the bucket of $bucket is not fully private: BlockPublicPolicy must be set to true" && exit 1
+  [ "${restrictPublicBuckets}" != "true" ] && echo "ERROR: the bucket of $bucket is not fully private: RestrictPublicBuckets must be set to true" && exit 1
+done
+
+echo "all buckets are private"
```
Comment (julialeague) [inline web-api/verify-private-s3-buckets.sh]: @ericsorenson new line? 😸

File: web-client/src/presenter/actions/navigateToCaseDetailWithDocketNumberAction.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import { state } from 'cerebral';
+
+/**
+ * changes the route to view the case-detail of the caseId of props.caseId
```
Comment (julialeague) [inline web-client/src/presenter/actions/navigateToCaseDetailWithDocketNumberAction.js]: @ericsorenson Should this be docketNumber instead for the references to caseId in the comments?

File: shared/src/business/entities/cases/Case.test.js (hunk 1 above)
Comment (ericsorenson) [reply shared/src/business/entities/cases/Case.test.js:141]: It could. Both point to the same value, so it would test the same thing. toRawObject strips out everything but the strings, numbers, booleans, etc.

File: shared/src/business/entities/cases/Case.test.js (hunk 2 above)
Comment (ericsorenson) [reply shared/src/business/entities/cases/Case.test.js]: Good catch, that was an accident!

File: shared/src/business/utilities/pdfGenerator/documentTemplates/NoticeOfTrialIssued.test.js (hunk 5 above)
Comment (ericsorenson) [reply shared/src/business/utilities/pdfGenerator/documentTemplates/NoticeOfTrialIssued.test.js:21]: They're almost the same, but not quite. Some use different props and functions.

File: shared/src/business/entities/cases/Case.test.js (hunk 1 above)
Comment (julialeague) [reply shared/src/business/entities/cases/Case.test.js:141]: `it('sets userId to current user if current user matches rawCase'` is what led me to think that `toRawObject` should be called. Is the referenced "rawCase" the same as the result of the raw object call?

File: shared/src/business/useCases/docketEntry/fileCourtIssuedDocketEntryInteractor.js (hunk 3 above)
Comment (ericsorenson) [reply shared/src/business/useCases/docketEntry/fileCourtIssuedDocketEntryInteractor.js:175]: Based on the coverage report, it looks like everything is covered apart from the following conditional date `filingDate: documentEntity.filingDate || createISODateString(),` which is no longer needed. I'll remove the default value.

File: web-client/src/presenter/actions/navigateToCaseDetailWithDocketNumberAction.js (hunk 8 above)
Comment (ericsorenson) [reply web-client/src/presenter/actions/navigateToCaseDetailWithDocketNumberAction.js]: Yes! Good catch!

File: shared/src/business/entities/cases/Case.test.js (hunk 2 above)
Comment (ericsorenson) [reply shared/src/business/entities/cases/Case.test.js]: Updated!

File: shared/src/business/utilities/htmlGenerator/index.pug (hunk 4 above)
Comment (ericsorenson) [reply shared/src/business/utilities/htmlGenerator/index.pug:16]: Updated!

File: web-api/verify-private-elasticsearch.sh (hunk 6 above)
Comment (ericsorenson) [reply web-api/verify-private-elasticsearch.sh]: Updated!

File: web-api/verify-private-s3-buckets.sh (hunk 7 above)
Comment (ericsorenson) [reply web-api/verify-private-s3-buckets.sh]: Updated!

File: web-client/src/presenter/actions/navigateToCaseDetailWithDocketNumberAction.js (hunk 8 above)
Comment (ericsorenson) [reply web-client/src/presenter/actions/navigateToCaseDetailWithDocketNumberAction.js]: Updated!

File: shared/src/business/entities/cases/Case.test.js (hunk 1 above)
Comment (ericsorenson) [reply shared/src/business/entities/cases/Case.test.js:141]: I see what you mean! I've updated the test description to be more accurate.

**Comment by julialeague** [comment]
@JessicaMarine , I recommend you accept this pull request into the staging environment. Thanks!
